core/src/main/java/org/elasticsearch/node/Node.java

```diff
@@ -282,28 +289,73 @@ public Node start() {
         injector.getInstance(MonitorService.class).start();
         injector.getInstance(RestController.class).start();
 
+        assert injector.getInstance(ClusterService.class) instanceof InternalClusterService :
+                "node cluster service implementation must inherit from InternalClusterService";
```

Comment:
We use it for testing. I'm on the path of indeed removing it, by making the internal cluster service be easier to instantiate - https://github.com/bleskes/elasticsearch/compare/discovery_guice...bleskes:decouple_cluster_service_transport